### PR TITLE
OTP-RR 3.0.0

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
This PR upgrades OTP-RR to 3.0.0. App.js and config.js are updated accordingly.
Most notably, all attributes in of the `configure()` function are now components (except `defaultMobileTitle`).

### CODE REVIEW

File: lib/app.js [hunk 1]
```diff
@@ -22,78 +17,47 @@ const otpConfig = require(YAML_CONFIG)
...
-            {getComponentIfExists(getContainerSlot)}
-            <Col sm={6} md={8} className='map-container'>
-              {getComponentIfExists(getMapContainerSlot)}
-              <Map />
-            </Col>
-          </Row>
-        </Grid>
-      </div>
-    )
+const TrimetWebapp = () => <ResponsiveWebapp components={components} />
```
Comment (binh-dam-ibigroup) [inline lib/app.js:61]: Should we just write `<ResponsiveWebapp components={jsConfig} />`?

File: lib/app.js [hunk 2]
```diff
@@ -22,78 +17,47 @@ const otpConfig = require(YAML_CONFIG)
...
+if (
+  !ItineraryBody ||
+  !LegIcon ||
+  !MainPanel ||
+  !MobileResultsScreen ||
+  !MobileSearchScreen ||
+  !ModeIcon
+) {
+  throw new Error('ItineraryBody, LegIcon, MainPanel, MobileResultsScreen, MobileSearchScreen, and ModeIcon must be defined in config.js')
 }
```
Comment (landonreed) [inline lib/app.js:44]:
```suggestion
const requiredComponents = {
  ItineraryBody,
  LegIcon,
  MainPanel,
  MobileResultsScreen,
  MobileSearchScreen,
  ModeIcon
}
const missingComponents = []
Object.keys(requiredComponents).forEach(key => {
  if (!requiredComponents[key]) missingComponents.push(key)
});

// Check that the required components are defined in the configuration.
if (missingComponents.length > 0) {
  throw new Error(`The following required components are missing from config.js: ${missingComponents.join(', ')}`)
}
```

**Review by landonreed — APPROVED** [review]
Looks good. Just one suggestion to simplify the checking of required components.

File: lib/app.js (hunk 1 above)
Comment (evansiroky) [reply lib/app.js:65]: I'd be fine with that if it works.

**Review by evansiroky — APPROVED** [review]

File: lib/app.js (hunk 1 above)
Comment (binh-dam-ibigroup) [reply lib/app.js]: Updated in https://github.com/ibi-group/trimet-mod-otp/pull/292/commits/5bc212d92aba21da438632e14aaa74dd78e7654f.

**Pull request merged**